kolibri/core/public/api.py

```diff
@@ -300,7 +303,14 @@ def create_or_update(self, request):
         user_id = serializer.validated_data["user"]
         instance_id = serializer.validated_data["instance"]
 
-        if not FacilityUser.objects.filter(id=user_id).exists():
+        if (
+            not FacilityUser.all_objects.filter(id=user_id).exists()
+            and not Store.objects.filter(id=user_id)
+            .filter(Q(deleted=True) | Q(hard_deleted=True))
+            .exists()
+            and not DeletedModels.objects.filter(id=user_id).exists()
+            and not HardDeletedModels.objects.filter(id=user_id).exists()
```

Comment:
The `Store` query and the `*DeletedModels` queries should be redundant. I would suggest just keeping the `*DeletedModels` ones because they're more compact.

Reply:
Apologies, I had this backwards in my mind. You're right, they're created when deletion occurs and deleted when serialization occurs. Now my comments make no sense